web/src/lib/components/album-page/albums-list.svelte

```diff
@@ -0,0 +1,287 @@
+  import AlbumsTable from '$lib/components/album-page/albums-table.svelte';
+  import { handleError } from '$lib/utils/handle-error';
+  import { searchAlbum } from './albums-controls.svelte';
+
+  export let albums: AlbumResponseDto[];
+
+  let shouldShowEditAlbumForm = false;
+  let selectedAlbum: AlbumResponseDto;
+  let albumToDelete: AlbumResponseDto | null;
+  let contextMenuPosition: OnShowContextMenuDetail = { x: 0, y: 0 };
+  let contextMenuTargetAlbum: AlbumResponseDto | undefined = undefined;
+
+  $: {
+    for (const key of sortByOptions) {
+      if (key.title === $albumViewSettings.sortBy) {
+        albums = key.sortFn(key.sortDesc, albums);
+        $albumViewSettings.sortDesc = key.sortDesc; // "Save" sortDesc
+        $albumViewSettings.sortBy = key.title;
+        break;
+      }
+    }
+  }
+  $: isShowContextMenu = !!contextMenuTargetAlbum;
+  $: albumsFiltered = albums.filter((album) => album.albumName.toLowerCase().includes($searchAlbum.toLowerCase()));
+
+  onMount(async () => {
+    await removeAlbumsIfEmpty();
+  });
+
+  function showAlbumContextMenu(contextMenuDetail: OnShowContextMenuDetail, album: AlbumResponseDto): void {
+    contextMenuTargetAlbum = album;
+    contextMenuPosition = {
+      x: contextMenuDetail.x,
+      y: contextMenuDetail.y,
+    };
+  }
+
+  function closeAlbumContextMenu() {
+    contextMenuTargetAlbum = undefined;
+  }
+
+  async function handleDeleteAlbum(albumToDelete: AlbumResponseDto): Promise<void> {
+    await deleteAlbum({ id: albumToDelete.id });
+    albums = albums.filter(({ id }) => id !== albumToDelete.id);
+  }
+
+  const chooseAlbumToDelete = (album: AlbumResponseDto) => {
+    contextMenuTargetAlbum = album;
+    setAlbumToDelete();
+  };
+
+  const setAlbumToDelete = () => {
+    albumToDelete = contextMenuTargetAlbum ?? null;
+    closeAlbumContextMenu();
+  };
+
+  const handleEdit = (album: AlbumResponseDto) => {
+    selectedAlbum = { ...album };
+    shouldShowEditAlbumForm = true;
+  };
+
+  const deleteSelectedAlbum = async () => {
+    if (!albumToDelete) {
+      return;
+    }
+    try {
+      await handleDeleteAlbum(albumToDelete);
+    } catch {
+      notificationController.show({
+        message: 'Error deleting album',
+        type: NotificationType.Error,
+      });
+    } finally {
+      albumToDelete = null;
+    }
+  };
+
+  const removeAlbumsIfEmpty = async () => {
+    try {
+      for (const album of albums) {
+        if (album.assetCount == 0 && album.albumName == '') {
+          await handleDeleteAlbum(album);
+        }
+      }
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  const successModifyAlbum = () => {
+    shouldShowEditAlbumForm = false;
+    notificationController.show({
+      message: 'Album infos updated',
+      type: NotificationType.Info,
+    });
+    albums[albums.findIndex((x) => x.id === selectedAlbum.id)] = selectedAlbum;
```

Comment:
Why not do an `album.find` in the first place?

Reply:
Because you need to replace the album in the array

Reply:
Oh yeah sorry that was stupid, my bad.

Reply:
Looking at it again this confuses me. We find the index i at which `selectedAlbum` is in the `albums` array. And then, we set `albums[i] = selectedAlbum`. But doesn't `albums[i]` already have the selected album? Otherwise, we wouldn't have found it in the first place, no?

Reply:
Ooooh. I finally got it, thanks! And sorry for all the explanation I needed lol